File: ios/Kinowo/Views/CinemaHeaderStyle.swift
import SwiftUI

/// Live-tunable look of the Kina tab's per-cinema section header and the gaps
/// around it, read by `CinemaSectionedGridView` from the environment.
///
/// Production never injects one, so the defaults below — which mirror the
/// literals `CinemaSectionedGridView` used before this struct existed —
/// reproduce the shipping Kina layout point for point. The non-prod tuning
/// pager (Kina page) swaps in an edited copy to preview header / spacing
/// changes against the real cinema-sectioned grid. Keep these defaults in
/// lockstep with the literals the grid would otherwise hard-code.
struct CinemaHeaderStyle: Equatable {
    /// Cinema section title font size (web `.cinema-section-title`).
    var fontSize: CGFloat = 15
    var fontWeight: Font.Weight = .semibold
    /// Thickness of the #3a3a6e underline beneath the title.
    var underlineThickness: CGFloat = 1
    /// Gap between the title text and its underline.
    var titleBottomPadding: CGFloat = 6
    /// `LazyVStack(spacing:)` between adjacent cinema sections.
    var sectionSpacing: CGFloat = 20
    /// Inner `VStack(spacing:)` between a section header and its film grid.
    var headerToGrid: CGFloat = 12
}

private struct CinemaHeaderStyleKey: EnvironmentKey {
    static let defaultValue = CinemaHeaderStyle()
}

extension EnvironmentValues {
    var cinemaHeaderStyle: CinemaHeaderStyle {
        get { self[CinemaHeaderStyleKey.self] }
        set { self[CinemaHeaderStyleKey.self] = newValue }
    }
}

File: android/app/src/main/java/pl/kinowo/ui/common/CinemaHeaderStyle.kt
package pl.kinowo.ui.common

import androidx.compose.runtime.compositionLocalOf
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Live-tunable look of the Kina-tab cinema section header (the blue cinema name
 * above each section's 2-column poster grid), read from [LocalCinemaHeaderStyle]
 * by `pl.kinowo.ui.list.CinemaGrid`.
 *
 * The defaults are the SHIPPING values, so production (which never provides its
 * own value) renders exactly as before. The non-prod tuning pager injects an
 * edited copy via `CompositionLocalProvider` to preview header changes against
 * the real sectioned list. Mirrors the role of [CardSpacingStyle] for the card.
 *
 * The live header is a plain blue `Text` with no underline, so there is no
 * underline lever — only font, weight, and the gaps around it.
 */
data class CinemaHeaderStyle(
    /** Cinema-name font size. */
    val fontSize: TextUnit = 15.sp,
    /** Cinema-name font weight. */
    val fontWeight: FontWeight = FontWeight.SemiBold,
    /** Gap above the header (between the previous section's last row and it). */
    val headerTopGap: Dp = 8.dp,
    /** Gap below the header, above its section's first poster row. */
    val headerToGrid: Dp = 2.dp,
    /** Vertical gap between stacked grid rows (the section-to-section rhythm). */
    val sectionSpacing: Dp = 12.dp,
)

/** Style driving every Kina-tab cinema header. The default equals today's
 *  shipping values, so production renders exactly as before; the tuning pager
 *  overrides it through `CompositionLocalProvider`. */
val LocalCinemaHeaderStyle = compositionLocalOf { CinemaHeaderStyle() }
